src/main/java/io/vertx/httpproxy/impl/ReverseProxy.java

```diff
@@ -191,27 +194,11 @@ public Future<Void> sendResponse() {
     }
 
     private Future<ProxyResponse> sendProxyRequest(ProxyRequest proxyRequest) {
-      Future<HttpClientRequest> f = resolveOrigin(proxyRequest.proxiedRequest());
-      f.onFailure(err -> {
-        // Should this be done here ? I don't think so
-        HttpServerRequest proxiedRequest = proxyRequest.proxiedRequest();
-        proxiedRequest.resume();
-        Promise<Void> promise = Promise.promise();
-        proxiedRequest.exceptionHandler(promise::tryFail);
-        proxiedRequest.endHandler(promise::tryComplete);
-        promise.future().onComplete(ar2 -> {
-          end(proxyRequest, 502);
-        });
-      });
-      return f.compose(a -> sendProxyRequest(proxyRequest, a));
+      return resolveOrigin(proxyRequest.proxiedRequest()).compose(a -> sendProxyRequest(proxyRequest, a));
     }
 
     private Future<ProxyResponse> sendProxyRequest(ProxyRequest proxyRequest, HttpClientRequest request) {
-      Future<ProxyResponse> fut = proxyRequest.send(request);
-      fut.onFailure(err -> {
-        proxyRequest.proxiedRequest().response().setStatusCode(502).end();
-      });
-      return fut;
+      return proxyRequest.send(request);
```

Comment:
It would be fine to inline this method (not reused elsewhere and the caller is pretty small).